[ao][sparsity] README for BaseDataSparsifier

macandro96 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82131
* __->__ #82130
* #82129

The readme file contains an overview of the base data sparsifier, it's implementation details.
Also, consists of code snippets and instructions on how to create your own custom
data sparsifier.

Test Plan: None

Commit:
[ao][sparsity] README for BaseDataSparsifier

The readme file contains an overview of the base data sparsifier, it's implementation details.
Also, consists of code snippets and instructions on how to create your own custom
data sparsifier.

Test Plan: None

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] README for BaseDataSparsifier"


The readme file contains an overview of the base data sparsifier, it's implementation details.
Also, consists of code snippets and instructions on how to create your own custom
data sparsifier.

Test Plan: None

[ghstack-poisoned]

z-a-f commented on torch/ao/sparsity/_experimental/data_sparsifier/README.md:
````diff
@@ -0,0 +1,128 @@
+data_sparsifier = ImplementedDataSparsifier()
+data_sparsifier.add_data(name=name, data=data, **some_config)
+```
+`step`: applies the update_mask() logic to all the data.
````
Blank spaces after a code block

z-a-f commented on torch/ao/sparsity/_experimental/data_sparsifier/README.md:
```diff
@@ -0,0 +1,128 @@
+If the named data already exists, then it is replaced with the new data. The config and mask will be retained for the new data unless not specified to.
+To not the old mask, set `use_old_mask=False`. If the `config` is explicitly passed in, it will be updated.
+
+**Note**: name containing '.' is not a valid name for the data sparsifier
```
nit: Paraphrase this, please: "Data sparsifier `name` argument cannot have '.' in it."

z-a-f commented on torch/ao/sparsity/_experimental/data_sparsifier/README.md:
```diff
@@ -0,0 +1,128 @@
+## API details
+`BaseDataSparsifier`: base class with abstract method `update_mask` that computes the new mask for all the data.
+
+`add_data`: Accepts name, data tuple and registers the data as a parametrized buffer inside the container model. Note that the data is always associated to a name. A custom sparse config can be provided along with the name, data pair. If not provided, the default config will be applied while doing the sparsification.
```
Can we make all methods a part of a bullet list?

z-a-f commented on torch/ao/sparsity/_experimental/data_sparsifier/README.md:
````diff
@@ -0,0 +1,128 @@
+
+## Using Data Sparsifier
+### Simple example
+```
````
blank spaces around the code blocks

z-a-f reviewed (approved):
Some nits -- land after addressing

Commit:
Update on "[ao][sparsity] README for BaseDataSparsifier"


The readme file contains an overview of the base data sparsifier, it's implementation details.
Also, consists of code snippets and instructions on how to create your own custom
data sparsifier.

Test Plan: None

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] README for BaseDataSparsifier"


The readme file contains an overview of the base data sparsifier, it's implementation details.
Also, consists of code snippets and instructions on how to create your own custom
data sparsifier.

Test Plan: None

[ghstack-poisoned]

macandro96 commented:
@pytorchbot merge -g
